inc/zoo/meta/BitmaskMaker.h

```diff
@@ -40,8 +40,20 @@ struct BitmaskMaker {
         >::value;
 };
 
+/// Provides TopBlit to clear any garbage bits at the top of a bitmask made by
+//BitmaskMaker
+template<typename T, int CurrentSize>
+struct BitmaskMakerClearTop {
+    constexpr static T BitCount = sizeof(T)*8;
+    constexpr static T LeastBitKeepCount = BitCount - BitCount % CurrentSize;
+    constexpr static T BitMod = BitCount % CurrentSize;
+    constexpr static T TopBlit = (BitMod == 0) ? ~(T(0)) :
```

Comment:
```suggestion
    constexpr static auto BitCountRemainder = BitCount % CurrentSize;
    constexpr static auto LeastBitsKeepCount = BitCount - BitCountRemainder;
    constexpr static auto TopBlitter =
        BitMod == 0 ?
            ~(T(0))  :
```

Reply:
It is really ugly the problem that shifts of exactly the register bit size are "undefined behavior":
[C++ standard draft $7.6.7 expr.shift](https://eel.is/c++draft/expr.shift#1)
I think the culprit is Intel's specification